core/Url.php

```diff
@@ -686,4 +676,31 @@ public static function getLocalHostnames()
     {
         return array('localhost', '127.0.0.1', '::1', '[::1]');
     }
+
+    /**
+     * @return string
+     */
+    public static function getCurrentSchemeFromRequestHeader()
+    {
+        if ((isset($_SERVER['HTTPS']) && ($_SERVER['HTTPS'] == 'on' || $_SERVER['HTTPS'] === true))
+            || (isset($_SERVER['HTTP_X_FORWARDED_PROTO']) && $_SERVER['HTTP_X_FORWARDED_PROTO'] == 'https')
+        ) {
+
+            return 'https';
+        }
+        return 'http';
+    }
+
+    /**
+     * @return bool
+     */
+    public static function isPiwikServerAssumeSecureConnectionIsUsed()
+    {
+        try {
+            $assume_secure_protocol = @Config::getInstance()->General['assume_secure_protocol'];
```

Comment:
Try/Catch might not be needed but can be ignored since it was already there before :)